cli/internal/env/env.go

```diff
@@ -26,15 +26,50 @@ type DetailedMap struct {
 	BySource BySource
 }
 
-// Merge takes another EnvironmentVariableMap and merges it into the receiver
-// It overwrites values if they already exist, but since the source of both will be os.Environ()
-// it doesn't matter
-func (evm EnvironmentVariableMap) Merge(another EnvironmentVariableMap) {
+// EnvironmentVariablePairs is a list of "k=v" strings for env variables and their values
+type EnvironmentVariablePairs []string
+
+// WildcardMaps is a pair of EnvironmentVariableMaps.
+type WildcardMaps struct {
+	Inclusions EnvironmentVariableMap
+	Exclusions EnvironmentVariableMap
+}
+
+// Resolve collapses a WildcardSet into a single EnvironmentVariableMap.
+func (ws WildcardMaps) Resolve() EnvironmentVariableMap {
+	output := EnvironmentVariableMap{}
+	output.Union(ws.Inclusions)
+	output.Difference(ws.Exclusions)
+	return output
+}
+
+// GetEnvMap returns a map of env vars and their values from os.Environ
+func GetEnvMap() EnvironmentVariableMap {
+	envMap := make(map[string]string)
+	for _, envVar := range os.Environ() {
+		if i := strings.Index(envVar, "="); i >= 0 {
```

Comment:
I didn't touch the implementation here, but since this ends up making a syscall and has platform specific behavior I _really_ don't mind it being defensive:
https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/syscall/env_unix.go;l=34